### DISCUSSION TITLE
Resize from any edge

### DISCUSSION THREAD

**Discussion by NicholasSloan** [rxhanson/RectanglePro-Community#107] [Ideas]
Hi Ryan,
Not wanting to gild the lily, but when resizing windows with a modifier drag, would it be too much to ask that the resizing should happen to the edge nearest the cursor? Most apps will allow you to drag any edge (though it can be tricky to find the exact spot) but RP drags only RH and bottom edges. Ideally with mod-drag I'd like it if I was acting on the edges nearest the cursor. Perhaps there is a reason why this would be difficult to implement?
Thanks

**Comment by NicholasSloan** [comment]
In anticipation of any questions about thresholds: the side of the window where a drag *begins* should determine the affected edges, though you might want to bias the position of the threshold cross slightly towards the existing/default behaviour. There might also be a case for zones where the drag would be restrict to one edge only —the pale green zones in the attached image, the red cross being the basic threshold. Probably you'd want a pref to choose between default and flexible behaviour?
Thanks
![Finder window+](https://user-images.githubusercontent.com/20109637/189514462-208e5924-4fdf-4052-aa1b-93cff0263ec4.png)

**Comment by rxhanson** [comment]
Thanks for the details. The reason that this isn't in the app is because resizing windows from a corner other than the bottom right is really janky using what's available to me as a 3rd party. The other window edges jump all around while resizing and I don't like it. I might add in a terminal command flag at some point for it, since this is commonly asked for.

**Comment by NicholasSloan** [comment]
Thanks for the explanation Ryan, I had a feeling there must be a good reason why this was not already implemented. Ah well…

**Comment by jennifer-richards** [comment]
Came to request this feature, which I have missed for many years (used to use it with twm, IIRC). Just discovered the modifiers for moving / resizing, and they're great! If the API ever changes and makes usable nearest-edge/corner resizing feasible, please +1 the request.

**Reply by rxhanson** [reply]
It seems quite unlikely that Apple would change this, but would you still be interested if the top left and bottom left corner resizing is imperfect? (Still gauging interest for it from a hidden config perspective).

**Reply by jennifer-richards** [reply]
Depending a bit on what "imperfect" means, very likely yes. My current workaround is to first move, then resize. That works well enough but requires a bit of planning. If the imperfect implementation is just a little non-smooth, that'd be fine.

What drives my feature request is being able to resize a window whose right or bottom edge (or both) is already in the right place. If "imperfect" preserves that, even if it's glitchy I would find it useful.

**Reply by sfgiants32** [reply]
+1 from me on this. I just started Rectangle Pro (not sure what took me so long to find), and I have this same use case ... love the quick resize feature but for windows in my bottom/right it doesn't work. I ahve to do the same - move the window first and then resize. Even if a bit bumpy/unsmooth, I'd take the option to resize from top-left and bottom-right corners.

**Reply by jennifer-richards** [reply]
FYI, I've found a pretty good workaround by mapping ctrl-cmd-mouse to move a window and ctrl-cmd-shift-mouse to resize. With a little practice this lets me switch between move/resize quite fluidly just by adding/removing the shift from the combination.

**Reply by rxhanson** [reply]
I'll give this another look in an upcoming release.

**Reply by Kater--S** [reply]
I came here for exactly the same question :)

I think Zooom2 had this feature and worked flawlessly (at least for me as a user). I loved and used it until it was discontinued. Resizing and zooming in one go (only pressing/releasing e.g. the alt key while still holding ctrl and cmd) without exact pointing and clicking is a bliss and a considerable saving of effort when handling your UI. I am also not used to tiling my windows as I prefer resizing them according to their actual content and also having some "air" around them, so all those features are less important to me.

Another point: Being able to resize and move a window in the background without bringing it to front (and into focus) would be great.
